src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,114 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.exceptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import com.internet.shop.util.ConnectionUtil;
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+    @Override
+    public Product create(Product product) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "INSERT INTO products (name, price) VALUES (?, ?);";
+            PreparedStatement statement =
+                    connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, product.getName());
+            statement.setBigDecimal(2, BigDecimal.valueOf(product.getPrice()));
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                product.setId(resultSet.getLong(1));
+            }
+            return product;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add " + product + " to the database.", e);
+        }
+    }
+
+    @Override
+    public Optional<Product> getById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM products WHERE product_id = ? AND deleted = false;";
+            PreparedStatement statement =
```

Comment:
```suggestion
            PreparedStatement statement = connection.prepareStatement(query);
```
can we inline the declaration and initialization here?